packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -127,3 +133,229 @@ export function useVolumeLevel(audioTrack?: IRemoteAudioTrack | ILocalAudioTrack
 
   return volumeLevel;
 }
+
+/**
+ * Auto-subscribe and get remote user audio track.
+ * Unsubscribe track on unmount.
+ */
+export function useRemoteAudioTracks(
+  users: IAgoraRTCRemoteUser[] | undefined,
+  client?: IAgoraRTCClient | null,
+): IRemoteAudioTrack[] {
+  const resolvedClient = useRTCClient(client);
+  const [tracks, setTracks] = useState<IRemoteAudioTrack[]>([]);
+  const isConnected = useIsConnected();
+  const nextTracks = useRef<IRemoteAudioTrack[]>([]);
+
+  useAsyncEffect(async () => {
+    if (!users || !isConnected) return;
+    let isUnmounted = false;
+    const subscribe = async (user: IAgoraRTCRemoteUser) => {
+      try {
+        if (!user.audioTrack && users.some(({ uid }) => user.uid === uid)) {
+          await resolvedClient.subscribe(user, "video").catch(error => {
+            void error;
+          });
+          if (
+            user.audioTrack &&
+            !nextTracks.current.some(track => track.getUserId() === user.uid)
+          ) {
+            nextTracks.current.push(user.audioTrack);
+          }
+
+          nextTracks.current = nextTracks.current.map(track => {
+            if (
+              user.audioTrack &&
+              track.getUserId() === user.uid &&
+              track.getTrackId() !== user.audioTrack.getTrackId()
+            ) {
+              return user.audioTrack;
+            } else {
+              return track;
+            }
+          });
```

Comment:
Maybe refactor to

```js
let newTracks = users.map(user => user.audioTrack).filter(Boolean)
if (!isSame(newTracks, nextTracks.current)) {
  setTracks(nextTracks.current = newTracks)
}
```
A more complex implementation see hooks/users.ts/usePublishedRemoteUsers.

Reply:
This is because the hot update will cause the track to be regenerated, so I used a comparison to ensure that the track I stored is up-to-date. 
At present, this way of writing only involves the calculation of an array space of nextTracks, and the complexity will be lower. Here I'm sorry for the complicated writing, but I want to keep it.